### ISSUE TITLE
[no-unused-vars] False positive: export with in keyof

### ISSUE THREAD

**Issue by OriginLive** [typescript-eslint/typescript-eslint#2615]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

// file 1
```ts
export type FieldDefinition = {
  [key: string]: string[];
};
```

// file 2
```ts
interface FruitsFormDefinition {
  apple: string;
  orange: string;
}

export type FruitsFieldDefinition = { [K in keyof FruitsFormDefinition]: FieldDefinition[string] };

export const fruitsFieldDefiniton: FruitsFieldDefinition = {
  apple: [],
  orange: ['isRequired'],
};
```

tsconfig:
```
{
  "compilerOptions": {
    "target": "es5",
    "lib": [
      "dom",
      "dom.iterable",
      "esnext"
    ],
    "allowJs": false,
    "skipLibCheck": false,
    "esModuleInterop": true,
    "allowSyntheticDefaultImports": true,
    "strict": true,
    "forceConsistentCasingInFileNames": true,
    "module": "esnext",
    "moduleResolution": "node",
    "resolveJsonModule": true,
    "isolatedModules": true,
    "noImplicitAny": true,
    "downlevelIteration": true,
    "noEmit": true,
    "jsx": "react",
    "typeRoots": [
      "./node_modules/@types",
      "./src/@types"
    ]
  },
  "include": [
    "src"
  ],
  "exclude": [
    "config-overrides.js"
  ]
}

```

**Expected Result**

no errors

**Actual Result**

Line 32:37: 'K' is defined but never used. Allowed unused vars must match /^_/u @typescript-eslint/no-unused-vars

**Additional Info**

Also getting is already declared in the upper scope                 no-shadow
not sure it's related or might go away with the fix

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.3.0` |
| `@typescript-eslint/parser`        | `4.3.0` |
| `TypeScript`                       | `4.0.3` |

**Comment by OriginLive** [comment]
another example of same error
```ts
export type FormValues<T> = {
  [K in keyof T]: FormValue;
};
```

**Comment by yeonjuan** [comment]
@OriginLive Hi :) I cannot reproduce it. Could you share your eslint config file or way to repro it?

**Comment by bradzacher** [comment]
This does not repro against master.
Likely you're using an old version of our tooling.

https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md#how-do-i-check-to-see-what-versions-are-installed

**Comment by OriginLive** [comment]
just tried it again today and it seems to work. No idea what changed
